Fix vars plugin code and documentation mismatch (#78562)
ansible/ansible: docs/docsite/rst/dev_guide/developing_plugins.rst

Since Ansible version 2.4, vars plugins only execute as needed when preparing to execute a task. This avoids the costly 'always execute' behavior that occurred during inventory construction in older versions of Ansible. Since Ansible version 2.10, vars plugin execution can be toggled by the user to run when preparing to execute a task or after importing an inventory source.

> Vars plugins in collections are required to be enabled by the user.

Reviewer: ```suggestion
The user must enable vars plugins in collections .
```

Reviewer 2: I would rephrase and link the config:
```
- The user must explicitly enable vars plugins that reside in a collection. See :ref:`enabled_vars_plugins` for details.
```

Author: ```suggestion
The user must explicitly enable vars plugins that reside in a collection :ref:`enable_vars`.
```

> The user must explicitly enable vars plugins that reside in a collection :ref:`enable_vars`.

Author: ```suggestion
The user must explicitly enable vars plugins that reside in a collection. See :ref:`enable_vars` for details.
```
